### PR TITLE
Fixed imports and tox configuration per isort 5.

### PR DESCRIPTION
**Pull request by smithdc1**
--recursive option has been removed. Directories passed in are now automatically sorted recursive.

isort version 5 has been released and it's broken the CI. I think this is the fix. 
 [CHANGELOG](
https://github.com/timothycrosley/isort/blob/develop/CHANGELOG.md)

### CODE REVIEW

**Comment by felixxm** [comment]
@smithdc1 I turned off `--recursive` in CI. Would you like to fix all `isort >= 5.0.0` failures?

**Comment by smithdc1** [comment]
Thank you. 

I've got some time this evening so let me have a deeper look.

**Comment by felixxm** [comment]
Thanks. At first glance, these failures seem reasonable.

File: tox.ini [hunk 1]
```diff
@@ -55,7 +55,7 @@ basepython = python3
 usedevelop = false
 deps = isort
 changedir = {toxinidir}
-commands = isort --recursive --check-only --diff django tests scripts
+commands = isort --check-only --diff django tests scripts
```
Comment (felixxm) [inline tox.ini:58]: We should pin `isort` version:
```diff
diff --git a/tox.ini b/tox.ini
index 8ead652d9d..79c2ff1247 100644
--- a/tox.ini
+++ b/tox.ini
@@ -9,7 +9,7 @@ envlist =
     py3
     flake8
     docs
-    isort
+    isort >= 5.0.0
 
 # Add environment to use the default python3 installation
 [testenv:py3]
```

**Comment by smithdc1** [comment]
@felixxm 

Is there any appetite to increase line length as 79 chars is enforced now. If not up to 119 we could go up to 88 (+10%). Here is are a couple of examples. The other alternative would be # isort : skip.

Final thing for now is I think I've found a line space bug. isort keeps adding line spaces in some circumstances. I'll carry on working on this see if I can find a minimal working solution to go raise an issue. 

**OLD**
``` python
from django.contrib.gis.db.backends.oracle.models import OracleGeometryColumns
```


**New**
``` python
from django.contrib.gis.db.backends.oracle.models import (
    OracleGeometryColumns,
)
```

or

**OLD**
``` python
            from django.db.models.sql.query import Query  # avoid circular import
```

**NEW**
``` python
            from django.db.models.sql.query import \
                Query,  # avoid circular import
```

**Comment by felixxm** [comment]
> from django.contrib.gis.db.backends.oracle.models import OracleGeometryColumns

**Comment by felixxm** [comment]
> Is there any appetite to increase line length as 79 chars is enforced now. If not up to 119 we could go up to 88 (+10%). Here is are a couple of examples.

No, this PR should be limited to fixing failures without changing our coding style. It seems that `isort` started to complain about inner imports, IMO it's fine to fix them.

**Comment by felixxm** [comment]
>             from django.db.models.sql.query import \
>                Query,  # avoid circular import

That's weird :thinking:

**Comment by smithdc1** [comment]
Yes it's odd. The comment is 2 chars over the line length and it's caused a wrap. 

I'm struggling for a minimal solution for the line space issue.
All I have done is run isort once (in theory that fixes it). But then it fails the check as it wants to add even more blank lines. 

Here is an example
```
--- C:\Users\smith\PycharmProjects\django2\django\core\management\commands\diffsettings.py:before       2020-07-04 21:29:01.856110
+++ C:\Users\smith\PycharmProjects\django2\django\core\management\commands\diffsettings.py:after        2020-07-04 21:33:51.565138
@@ -42,6 +42,7 @@
         from django.conf import Settings, global_settings, settings


+
         # Because settings are imported lazily, we need to explicitly load them.
         if not settings.configured:
             settings._setup()
ERROR: C:\Users\smith\PycharmProjects\django2\django\core\management\commands\shell.py Imports are incorrectly sorted and/or formatted.
```

**Comment by felixxm** [comment]
So we have at least one bug in `isort` 5+:
```python
from django.db.models.sql.query import \
    Query,  # avoid circular import
```
I will pin `isort < 5` in CI for now. We need to wait for a new release.

**Comment by smithdc1** [comment]
@felixxm ok. That sounds good. It seems to be to be a bit unstable to me. I think I have a particular file fixed, commit the change and then it fails again. (but then again it could just be me...)

Enjoy the rest of your weekend 👍

**Comment by felixxm** [comment]
@smithdc1 Thanks :+1: Let's wait for bugfixes in `isort`.

**Pull request closed without merging**